src/mongoc/mongoc-topology-private.h

```diff
@@ -59,7 +61,8 @@ typedef struct _mongoc_topology_t
 } mongoc_topology_t;
 
 mongoc_topology_t *
-mongoc_topology_new (const mongoc_uri_t *uri);
+mongoc_topology_new (const mongoc_uri_t *uri,
+                     bool                single_threaded);
```

Comment:
I think it makes more sense for the topology to be created in single/multi threaded mode and for it to own it's background thread